netbox/dcim/forms/filtersets.py

```diff
@@ -1461,6 +1461,11 @@ class InventoryItemFilterForm(DeviceComponentFilterForm):
             choices=BOOLEAN_WITH_BLANK_CHOICES
         )
     )
+    device_status = forms.MultipleChoiceField(
```

Comment:
Likewise, this can be moved to `DeviceComponentFilterForm`. We'll need to add the field to a fieldset in each of the individual components forms though.